test/controllers/records_controller_test.rb

```diff
@@ -121,6 +121,26 @@ class RecordsControllerTest < ActionController::TestCase
       assert_redirected_to collection_path(id: 1, record_id: record.id)
     end
 
+    it 'displays a failure message when attempting to associate a granule to a collection record that has been closed.' do
+      user = User.find_by(email: 'abaker@element84.com')
+      sign_in(user)
+      stub_urs_access(user.uid, user.access_token, user.refresh_token)
+
+      Record.any_instance.stubs(close!: true)
+      Record.stubs(:find).with(1).returns(Record.new(id: 1))
+
+      Record.stubs(:find_by).with(id: 'non_existent_granule_id').returns(nil)
```

Comment:
I commented on this yesterday but I don't see my comment.   I think you are stubbing too much here.   The test logic should flow like this:
1) Look at the fixtures, find a collection record id that is closed
2) Do your post and assoc a granule to that collection (you may need to add a fixture for the granule).
3) Verify the error